### ISSUE TITLE
[Bug]: clicking on etherscan's "Connect to web3" always fail 

### ISSUE THREAD

**Issue by zhiqiangxu** [MetaMask/metamask-extension#24388]
### Describe the bug

After opening [this page](https://sepolia.etherscan.io/address/0x530d072cB934b84f710Fe50F453A6e1e1d0fE75F#writeContract) and switching to the "Write Contract" tab, I click on the "Connect to web3" button, it instantly reports:

```
{code: 4902, message: 'Unrecognized chain ID. Try adding the chain using `wallet_addEthereumChain` first.'}
```

This problem is gone if I do it on mainnet, but I'm sure the sepolia network is already added to the list.

(Maybe it's more proper to fire this issue to etherscan, but it seems etherscan is not open source...)

### Expected behavior

"Connect to web3" should work as expected.

### Screenshots/Recordings

<img width="1669" alt="image" src="https://github.com/MetaMask/metamask-extension/assets/1265027/3513a7f5-6a8e-411f-a04e-de2f00c780d4">

### Steps to reproduce

Already described above.

### Version

11.14.5

### Build type

None

### Browser

Chrome

### Operating system

MacOS

**Comment by digiwand** [comment]
Hi @zhiqiangxu, thanks for reporting this issue.

Question: Is this bug still occurring if you manually switch to the Sepholia network in MetaMask before connecting? would you mind providing the RPC network / chain ID you were using at the time of the screenshot?

**Comment by zhiqiangxu** [comment]
Yes, it's still occurring even if I manually switch to the Sepholia network in MetaMask before connecting.

Another side problem is that the toggle button doesn't work neither.

This is what shows when I click on the sepolia network:

<img width="1418" alt="image" src="https://github.com/MetaMask/metamask-extension/assets/1265027/b3947ab1-f3a7-4e11-a992-42f3d3aef83e">


So it seems the network information for sepolia is somehow wrong? I can't manually edit it since it's locked.

**Comment by vandan** [comment]
Confirmed that this error seems to be reproducible in 11.16.14, but appears resolved in upcoming 12.0.0 builds.

**Comment by vandan** [comment]
Seems to be resolved on 12.0.5 please let us know if you encounter this issue on that version or higher.

**Comment by hjetpoluru** [comment]
The bug has been fixed and here is video attached. https://github.com/user-attachments/assets/5114b72a-a749-422a-9cde-76fffec89d33
